### PR TITLE
resource_manager: add the runaway field for resource group

### LINKED ISSUE #6474: Add a runaway field for resource group
## Development Task
ref https://github.com/pingcap/tidb/issues/43691
Add a runaway field for resource group

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: close #6474

### What is changed and how does it work?
 add the runaway field for resource group

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by Connor1996 — COMMENTED** [review]
Where is the parser change?

File: pkg/mcs/resource_manager/server/resource_group.go [hunk 1]
```diff
@@ -33,8 +33,9 @@ type ResourceGroup struct {
 	Name string         `json:"name"`
 	Mode rmpb.GroupMode `json:"mode"`
 	// RU settings
-	RUSettings *RequestUnitSettings `json:"r_u_settings,omitempty"`
-	Priority   uint32               `json:"priority"`
+	RUSettings *RequestUnitSettings  `json:"r_u_settings,omitempty"`
+	Priority   uint32                `json:"priority"`
+	Runaway    *rmpb.RunawaySettings `json:"runaway,omitempty"`
```
Comment (nolouch) [inline pkg/mcs/resource_manager/server/resource_group.go:38]:
```suggestion
	Runaway    *rmpb.RunawaySettings `json:"runaway_settings,omitempty"`
```

**Review by nolouch — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**